pkg/util/util_linux.go

```diff
@@ -16,12 +16,29 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package main
+package util
 
 import (
+	"fmt"
+	"os"
+
 	"golang.org/x/sys/unix"
 )
 
-func umask(mask int) (int, error) {
+func Umask(mask int) (int, error) {
 	return unix.Umask(mask), nil
 }
+
+func CleanupSocketFile(socketPath string) error {
+	fi, err := os.Stat(socketPath)
+	if err == nil && (fi.Mode()&os.ModeSocket) != 0 {
```

Comment:
FYI this might be buggy: https://github.com/kubernetes-csi/node-driver-registrar/pull/61#discussion_r377326338

We should return an error if the file isn't a socket. But yeah can be a separate PR

Reply:
@Madhu-1 FYI. IMO this can be reused in the deletion goroutine. We don't *need* the socket check there but it doesn't hurt. If another file type is there, the socket is gone anyway.

Reply:
Thank you, @verult . Will raise a shorter PR with that change right after this one gets merged.

Reply:
For the umask will follow up with a short PR removing the checks, since that is existing logic being modified.